### PR TITLE
decode message in socks only once

### PR DESCRIPTION
**Pull request by shrenujb**
### Changelist
Currently we assume a 1-1 relationship with the kafka topic and forwarded msg in socks. This changed with the introduction of the parent subaccounts channel which uses messages from the subaccounts topic. Currently, we're decoding the same kafka msg twice once for each channel. This change fixes that

### Test Plan
Existing tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **Refactor**
	- Improved message handling logic to support processing and forwarding multiple messages.

### CODE REVIEW

File: indexer/services/socks/__tests__/lib/message-forwarder.test.ts [hunk 1]
```diff
@@ -181,6 +181,7 @@ describe('message-forwarder', () => {
 
   beforeAll(async () => {
     await dbHelpers.migrate();
+    await dbHelpers.clearData();
```
Comment (Christopher-Li) [inline indexer/services/socks/__tests__/lib/message-forwarder.test.ts:184]: nit: delete this

File: indexer/services/socks/src/lib/message-forwarder.ts [hunk 2]
```diff
@@ -155,6 +142,48 @@ export class MessageForwarder {
...
+    //       `${config.SERVICE_NAME}.message_time_since_received`,
+    //       startForwardMessage - Number(originalMessageTimestamp),
+    //       STATS_NO_SAMPLING,
+    //       {
+    //         topic,
+    //         event_type: String(message.headers?.event_type),
+    //       },
+    //     );
+    //   }
+    // }
```
Comment (Christopher-Li) [inline indexer/services/socks/src/lib/message-forwarder.ts:186]:
```suggestion

```
nit: delete

File: indexer/services/socks/src/helpers/from-kafka-helpers.ts [hunk 3]
```diff
@@ -103,6 +103,74 @@ export function getMessageToForward(
   }
 }
 
+export function getMessagesToForward(topic: string, message: KafkaMessage): MessageToForward[] {
+  // const channels: Channel[] = getChannels(topic);
```
Comment (Christopher-Li) [inline indexer/services/socks/src/helpers/from-kafka-helpers.ts:107]: nit: delete

File: indexer/services/socks/src/helpers/from-kafka-helpers.ts [hunk 4]
```diff
@@ -103,6 +103,72 @@ export function getMessageToForward(
...
+        subaccountNumber: subaccountMessage.subaccountId!.number,
+        contents: getParentSubaccountContents(subaccountMessage),
+        version: subaccountMessage.version,
+      }];
+    }
+    default:
+      throw new InvalidForwardMessageError(`Unknown topic: ${topic}`);
+  }
+
+  return [] as MessageToForward[];
```
[the remark this answers is not in the document]
Comment (dydxwill) [reply indexer/services/socks/src/helpers/from-kafka-helpers.ts:169]: +1

File: indexer/services/socks/src/helpers/from-kafka-helpers.ts [hunk 5]
```diff
@@ -103,6 +103,72 @@ export function getMessageToForward(
   }
 }
```
Comment (Christopher-Li) [inline indexer/services/socks/src/helpers/from-kafka-helpers.ts:104]: delete unused function above

File: indexer/services/socks/src/helpers/from-kafka-helpers.ts (hunk 4 above)
Comment (Christopher-Li) [inline indexer/services/socks/src/helpers/from-kafka-helpers.ts:169]: I dont think this return is necessary

**Review by Christopher-Li — APPROVED** [review]

**Comment by shrenujb** [comment]
@mergify-io backport release/indexer/v5.x

**Comment by shrenujb** [comment]
@mergifyio backport release/indexer/v5.x

**Comment by Christopher-Li** [comment]
https://github.com/Mergifyio backport release/indexer/v5.x

**Pull request merged**